staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -498,11 +503,11 @@ type namespacedName struct {
 // Watch implements storage.Interface.
 func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions) (watch.Interface, error) {
 	pred := opts.Predicate
-	// If the resourceVersion is unset, ensure that the rv
-	// from which the watch is being served, is the latest
+	// if the watch-list feature wasn't set and the resourceVersion is unset
```

Comment:
Just to be really on the safe side - let's just add:
```
if utilfeature.DefaultFeatureGate.Enabled(features.WatchList) && opts.SendinitialEvents != nil {
  opts.SendInitialEvents = nil
}
```

[I know that technically it should never happen due to our validation, but let's just be super-safe here.]